src/gui/gtk.c

```diff
@@ -3150,8 +3149,15 @@ static gboolean _scroll_wrap_scroll(GtkScrolledWindow *sw, GdkEventScroll *event
     GtkAdjustment *adj = gtk_scrolled_window_get_vadjustment(sw);
 
     const gint before = gtk_adjustment_get_value(adj);
-    gint value = before + increment*event->delta_y;
-    value -= value % increment;
+    gint value = before + increment*delta_y;
+
+    const gint delta = value % increment;
```

Comment:
The purpose of this code is to end up on an item boundary, even if previously you were between item boundaries (for example because you directly manipulated the scrollbar with the mouse). Easiest to see in duplicate manager. Another way of writing this is
`value = (value / increment) * increment;`
So it has nothing to do (in theory) with direction of movement. I'm not sure why it doesn't work for you :-(